phc/easy/query.py

```diff
@@ -84,26 +97,25 @@ def execute_dsl(
         }, all_results=True)
         """
         if all_results:
-            return recursive_execute_dsl(
-                {
-                    **query,
-                    "limit": [
-                        {"type": "number", "value": 0},
-                        # Make window size smaller than maximum to reduce
-                        # pressure on API
-                        {"type": "number", "value": int(MAX_RESULT_SIZE / 5)},
-                    ],
-                },
-                all_results,
-                auth_args,
-            )
-
-        # Scroll if limit is above MAX_RESULT_SIZE
-        limit = iter(query.get("limit", []))
-
-        lower = next(limit, {}).get("value", 0)
-        upper = next(limit, {}).get("value", 0)
-
-        scroll = True if upper - lower > MAX_RESULT_SIZE else all_results
-
-        return recursive_execute_dsl(query, scroll, auth_args)
+            with tqdm(total=MAX_RESULT_SIZE) as progress:
```

Comment:
What is the experience like for someone using the SDK in a non interactive environment?   Does this just add some noise to stdout?

Reply:
It handles an interactive text console quite well. If it's a non-interactive environment, then yes, it'll get a bit cluttered. I'll add an option to disable the progress indicator. I need to discuss the previous PR with @mjtieman first though.

Reply:
One thing we do with `pandas` is use `extras_require` in setup.py to make the install of it optional.  Then, if pandas is present, we expose those features.  Could maybe do the same to see if tqdm is installed, then use it, otherwise don't.